apps/spruce/src/pages/waterfall/WaterfallGrid.tsx

```diff
@@ -28,23 +35,36 @@ export const WaterfallGrid: React.FC = () => {
   return (
     <Container>
       <Row>
-        <div /> {/* Placeholder div for the build variant label column */}
-        {data.waterfall.versions.map(({ version }) =>
-          version ? <VersionLabel key={version.id} {...version} /> : null,
-        )}
+        <BuildVariantTitle />
+        <Versions data-cy="version-labels">
+          {data.waterfall.versions.map(({ inactiveVersions, version }, i) =>
+            version ? (
+              <VersionLabel key={version.id} {...version} />
+            ) : (
+              <InactiveVersion
+                key={`${inactiveVersions?.[0]?.id}-${i}`} // eslint-disable-line react/no-array-index-key
```

Comment:
would just the ID be enough? (assuming inactive versions don't contain duplicate versions)
```suggestion
                key={inactiveVersions?.[0]?.id}
```

Reply:
I'll use it just as a fallback!